crates/cargo-contract/src/cmd/extrinsics/instantiate.rs

```diff
@@ -266,13 +266,33 @@ impl Exec {
             .find_first::<api::contracts::events::CodeStored>()?
             .map(|code_stored| code_stored.code_hash);
 
-        let instantiated = result
-            .find_first::<api::contracts::events::Instantiated>()?
-            .ok_or_else(|| anyhow!("Failed to find Instantiated event"))?;
+        // //TODO:Remove the useless part
+        // let instantiated = result
+        //     .find_first::<api::contracts::events::Instantiated>()?
+        //     .ok_or_else(|| anyhow!("Failed to find Instantiated event"))?;
+        let last_instantiated = result
+            .iter()
+            .filter_map(|ev| {
+                ev.and_then(|ev| {
+                    ev.as_event::<api::contracts::events::Instantiated>()
+                        .map_err(Into::into)
+                })
+                .transpose()
+            })
+            .last()
+            .transpose()?
+            .ok_or_else(|| anyhow!("Failed to find Last Instantiated event"))?;
```

Comment:
```suggestion
            .ok_or_else(|| anyhow!("Failed to find Instantiated event"))?;
```
I think this case is it failed to find *any* instantiated event